pkgs/by-name/tr/tribler/package.nix

```diff
@@ -1,131 +1,144 @@
 {
   lib,
   stdenv,
-  fetchurl,
-  fetchPypi,
-  python311,
+  python312,
   makeWrapper,
-  libtorrent-rasterbar-1_2_x,
-  qt5,
   nix-update-script,
-  boost186,
+  fetchFromGitHub,
+  rustPlatform,
+  buildNpmPackage,
+  nodejs_20,
 }:
-
 let
-  # libtorrent-rasterbar-1_2_x requires python311 and boost 1.86
-  python3 = python311.override {
-    packageOverrides = final: prev: {
-      boost = boost186;
+  version = "8.1.0";
+  python3 = python312;
+  nodejs = nodejs_20;
+
+  src = fetchFromGitHub {
+    rev = "v${version}";
+    owner = "tribler";
+    repo = "tribler";
+    hash = "sha256-hQ9O/ZqE7ZTduaFfdvS3szaJGwrv0Ch/HlcwMW23PPM=";
+  };
+
+  ipv8-rust-tunnels = python3.pkgs.buildPythonPackage rec {
```

Comment:
You can optionally put this into a seperate .nix file and then use callPackage to load it

So this file is shorter